internal/tools/pipeline/providers/dbgc/dbgc.go

```diff
@@ -61,33 +61,30 @@ func (p *provider) PipelineDatabaseGC(ctx context.Context) {
 }
 
 // doPipelineDatabaseGC query the data in the database according to req paging to perform gc
-func (p *provider) doPipelineDatabaseGC(req apistructs.PipelinePageListRequest) {
-	var pageNum = req.PageNum
-	pipelines := make([]spec.Pipeline, 0)
-	for {
-		req.PageNum = pageNum
+func (p *provider) doPipelineDatabaseGC(ctx context.Context, req apistructs.PipelinePageListRequest) {
+	rutil.ContinueWorking(ctx, p.Log, func(ctx context.Context) rutil.WaitDuration {
 		result, err := p.dbClient.PageListPipelines(req)
-		pageNum += 1
 		if err != nil {
 			p.Log.Errorf("failed to compensate pipeline req: %v, err: %v", req, err)
-			continue
+			return rutil.ContinueWorkingImmediately
```

Comment:
should sleep, or you may occur error infinitely

Reply:
adjusted, sleep with default interval